lib/src/util/context.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:meta/meta.dart';
+import 'package:over_react/src/component_declaration/component_base.dart' as component_base;
+import 'package:over_react/src/component_declaration/builder_helpers.dart' as builder_helpers;
+import 'package:over_react/over_react.dart';
+import 'package:react/react_client/react_interop.dart';
+import 'package:react/react_client/js_backed_map.dart';
+import 'package:react/react.dart' as react;
+
+class Context {
+  Context(this.Provider, this.Consumer, this.reactDartContext);
+
+  final react.ReactDartContext reactDartContext;
+
+  /// Every [Context] object comes with a Provider component that allows consuming components to subscribe
+  /// to context changes.
+  ///
+  /// Accepts a `value` prop to be passed to consuming components that are descendants of this [Provider].
+  ///
+  /// All consumers that are descendants of a Provider will re-render whenever the Provider’s value prop changes.
+  /// The propagation from Provider to its descendant consumers is not subject to the shouldComponentUpdate method,
+  /// so the consumer is updated even when an ancestor component bails out of the update.
+  ///
+  /// __Example__:
+  ///
+  ///     Context MyContext = createContext();
+  ///     ...
+  ///     class MyComponent extends UiComponent2<...> {
+  ///       render() {
+  ///         return (MyContext.Provider()..value = 'new context value')(
+  ///           MyContext.Consumer()(
+  ///             (value) {
+  ///               return Dom.span()(
+  ///                 '$value', // Outputs: 'new context value'
+  ///               );
+  ///             }
+  ///           );
+  ///         );
+  ///       }
+  ///     }
+  ///
+  /// Learn More: <https://reactjs.org/docs/context.html#contextprovider>
+  @override
+  final UiFactory<ProviderProps> Provider;
+
+  /// A React component that subscribes to context changes.
+  /// Requires a function as a child. The function receives the current context value and returns a React node.
+  ///
+  /// A UiComponent that subscribes to context changes.
+  /// This lets you subscribe to a context.
```

Comment:
Lines 45 - 46 seem to be pretty much that same as 48 - 49. Are all of those lines supposed to be there?